pkg/kubelet/kubelet_pods_test.go

```diff
@@ -179,7 +182,60 @@ func TestMakeMounts(t *testing.T) {
 	}
 }
 
-func TestHostsFileContent(t *testing.T) {
+func TestNodeHostsFileContent(t *testing.T) {
+	testCases := []struct {
+		hostsFileName   string
+		expectedContent string
+	}{
+		{
+			"hosts_test_file1",
+			`# hosts file for testing.
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+fe00::0	ip6-mcastprefix
+fe00::1	ip6-allnodes
+fe00::2	ip6-allrouters
+123.45.67.89	some.domain
+`,
+		},
+		{
+			"hosts_test_file2",
+			`# another hosts file for testing.
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+fe00::0	ip6-mcastprefix
+fe00::1	ip6-allnodes
+fe00::2	ip6-allrouters
+12.34.56.78	another.domain
+`,
+		},
+	}
+
+	for _, testCase := range testCases {
+		hostsFilePath := path.Join("/", "tmp", testCase.hostsFileName)
```

Comment:
Suggestion: use `ioutil.TempDir` and `ioutil.WriteFile` to simplify the code and write to an OS specific temp directory.
Example: https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/pkg/kubelet/dockershim/helpers_test.go#L161